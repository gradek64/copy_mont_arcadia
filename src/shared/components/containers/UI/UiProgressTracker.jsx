import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { addIndex, map, keys } from 'ramda'

import ProgressTracker from '../../common/ProgressTracker/ProgressTracker'

const EXAMPLES = {
  'CheckoutV1 3 Steps': ['Delivery', 'Billing', 'Confirm'],
  'CheckoutV2 3 Steps': ['Delivery', 'Payment', 'Thank You'],
  'CheckoutV2 4 Steps': ['Login', 'Delivery', 'Payment', 'Thank You'],
  'CheckoutV2 D&P 2 Steps': ['Delivery and Payment', 'Thank You'],
  'Example 4 Steps': ['Step0', 'Step1', 'Step2', 'Step3'],
}

const ACTIVE_STEP_OPTIONS = ['none', 0, 1, 2, 3]

export default class UiProgressTracker extends Component {
  static contextTypes = {
    l: PropTypes.func,
  }

  constructor(props) {
    super(props)
    this.state = {
      steps: EXAMPLES['CheckoutV1 3 Steps'],
      activeStepIdx: 'none',
    }
  }

  enrichSteps = (steps, activeStepIdx) => {
    const { l } = this.context
    return addIndex(map)((title, idx) => {
      return {
        title: l(title),
        active: `${idx}` === activeStepIdx,
      }
    }, steps)
  }

  onSelectActiveStep = (event) => {
    this.setState({ activeStepIdx: event.target.value })
  }

  onSelectExample = (event) => {
    this.setState({ steps: EXAMPLES[event.target.value] })
  }

  renderOption = (value) => {
    return (
      <option key={value} value={value}>
        {value}
      </option>
    )
  }

  render() {
    const steps = this.enrichSteps(this.state.steps, this.state.activeStepIdx)

    return (
      <div>
        <div style={{ border: '1px dotted grey', padding: '0 10px' }}>
          <ProgressTracker steps={steps} />
        </div>
        <div>
          <div className="Select is-selected">
            <div className="Select-head">
              <label className="Select-label" htmlFor="examples">
                ProgressTracker example:{' '}
              </label>
            </div>
            <div className="Select-container">
              <select
                className="Select-select"
                name="examples"
                onChange={this.onSelectExample}
              >
                {map(this.renderOption, keys(EXAMPLES))}
              </select>
            </div>
          </div>
          <div className="Select is-selected">
            <div className="Select-head">
              <label className="Select-label" htmlFor="indexes">
                ProgressTracker active step:{' '}
              </label>
            </div>
            <div className="Select-container">
              <select
                className="Select-select"
                name="indexes"
                onChange={this.onSelectActiveStep}
                value={this.state.activeStepIdx}
              >
                {map(this.renderOption, ACTIVE_STEP_OPTIONS)}
              </select>
            </div>
          </div>
        </div>
      </div>
    )
  }
}
