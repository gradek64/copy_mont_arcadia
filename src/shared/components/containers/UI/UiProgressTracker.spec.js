import testComponentHelper from 'test/unit/helpers/test-component'
import UiProgressTracker from './UiProgressTracker'

describe('<UiProgressTracker />', () => {
  describe('@renders', () => {
    const renderComponent = testComponentHelper(UiProgressTracker)
    it('in default state', () => {
      expect(renderComponent().getTree()).toMatchSnapshot()
    })
  })

  describe('@events', () => {
    const renderComponent = testComponentHelper(UiProgressTracker)
    describe('onSelectExample', () => {
      it('select the correct example', () => {
        const event = { target: { value: 'Example 4 Steps' } }
        const { wrapper, instance } = renderComponent()
        wrapper.find('select[name="examples"]').simulate('change', event)
        expect(instance.state.steps).toEqual([
          'Step0',
          'Step1',
          'Step2',
          'Step3',
        ])
      })
    })
    describe('onSelectActiveStep', () => {
      it('select the correct index', () => {
        const event = { target: { value: 2 } }
        const { wrapper, instance } = renderComponent()
        wrapper.find('select[name="indexes"]').simulate('change', event)
        expect(instance.state.activeStepIdx).toEqual(event.target.value)
      })
    })
  })
})
