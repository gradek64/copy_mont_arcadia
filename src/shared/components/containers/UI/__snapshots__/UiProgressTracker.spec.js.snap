// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UiProgressTracker /> @renders in default state 1`] = `
<div>
  <div
    style={
      Object {
        "border": "1px dotted grey",
        "padding": "0 10px",
      }
    }
  >
    <ProgressTracker
      steps={
        Array [
          Object {
            "active": false,
            "title": "Delivery",
          },
          Object {
            "active": false,
            "title": "Billing",
          },
          Object {
            "active": false,
            "title": "Confirm",
          },
        ]
      }
    />
  </div>
  <div>
    <div
      className="Select is-selected"
    >
      <div
        className="Select-head"
      >
        <label
          className="Select-label"
          htmlFor="examples"
        >
          ProgressTracker example:
           
        </label>
      </div>
      <div
        className="Select-container"
      >
        <select
          className="Select-select"
          name="examples"
          onChange={[Function]}
        >
          <option
            key="CheckoutV1 3 Steps"
            value="CheckoutV1 3 Steps"
          >
            CheckoutV1 3 Steps
          </option>
          <option
            key="CheckoutV2 3 Steps"
            value="CheckoutV2 3 Steps"
          >
            CheckoutV2 3 Steps
          </option>
          <option
            key="CheckoutV2 4 Steps"
            value="CheckoutV2 4 Steps"
          >
            CheckoutV2 4 Steps
          </option>
          <option
            key="CheckoutV2 D&P 2 Steps"
            value="CheckoutV2 D&P 2 Steps"
          >
            CheckoutV2 D&P 2 Steps
          </option>
          <option
            key="Example 4 Steps"
            value="Example 4 Steps"
          >
            Example 4 Steps
          </option>
        </select>
      </div>
    </div>
    <div
      className="Select is-selected"
    >
      <div
        className="Select-head"
      >
        <label
          className="Select-label"
          htmlFor="indexes"
        >
          ProgressTracker active step:
           
        </label>
      </div>
      <div
        className="Select-container"
      >
        <select
          className="Select-select"
          name="indexes"
          onChange={[Function]}
          value="none"
        >
          <option
            key="none"
            value="none"
          >
            none
          </option>
          <option
            key="0"
            value={0}
          >
            0
          </option>
          <option
            key="1"
            value={1}
          >
            1
          </option>
          <option
            key="2"
            value={2}
          >
            2
          </option>
          <option
            key="3"
            value={3}
          >
            3
          </option>
        </select>
      </div>
    </div>
  </div>
</div>
`;
